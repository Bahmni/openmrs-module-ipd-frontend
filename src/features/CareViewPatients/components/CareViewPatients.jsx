import React from "react";
import "../styles/CareViewPatients.scss";
import { Accordion, AccordionItem, Dropdown } from "carbon-components-react";

export const CareViewPatients = () => {
  return (
    <div className="care-view-patients-container">
      <div className="care-view-patients">
        <div className="task-type">
          <Dropdown
            id="default"
            label="Dropdown menu options"
            items={["All tasks", "Pending", "Done"]}
            itemToString={(item) => (item ? item : "")}
            selectedItem={"All tasks"}
          />
        </div>
        <Accordion className="patient-section">
          <AccordionItem title="Patient 1">Patient 1</AccordionItem>
          <AccordionItem title="Patient 2">Patient 2</AccordionItem>
        </Accordion>
      </div>
    </div>
  );
};
