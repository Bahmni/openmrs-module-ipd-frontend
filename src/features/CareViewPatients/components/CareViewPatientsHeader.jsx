import React from "react";
import PropTypes from "prop-types";
import "../styles/CareViewPatientsHeader.scss";
import { IPD_WARD_SEARCH_PLACEHOLDER_TEXT } from "../../../constants";
import { Dropdown, Search } from "carbon-components-react";

export const CareViewPatientsHeader = (props) => {
  const { handleKeyPress, handleClear, searchValue, updateSearchValue } = props;

  const handleSearchOnChange = (e) => {
    updateSearchValue(e.target.value);
  };

  return (
    <div className="task-type">
      <Search
        placeholder={IPD_WARD_SEARCH_PLACEHOLDER_TEXT}
        size="lg"
        id="ipd-ward-search"
        onChange={handleSearchOnChange}
        value={searchValue}
        onKeyDown={handleKeyPress}
        onClear={handleClear}
        className="ward-patient-search"
        labelText="Patient Search"
        light={true}
      />
      <Dropdown
        id="default"
        label="Dropdown menu options"
        items={["All tasks", "Pending", "Done"]}
        itemToString={(item) => (item ? item : "")}
        selectedItem={"All tasks"}
        light={true}
      />
    </div>
  );
};

CareViewPatientsHeader.propTypes = {
  handleKeyPress: PropTypes.func,
  handleClear: PropTypes.func,
  searchValue: PropTypes.string,
  updateSearchValue: PropTypes.func,
};
