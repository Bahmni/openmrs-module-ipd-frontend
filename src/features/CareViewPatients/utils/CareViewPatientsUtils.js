import axios from "axios";
import {
  BOOKMARK_PATIENT_BASE_URL,
  GET_PATIENT_LIST_URL,
  GET_SEARCH_PATIENT_LIST_URL,
} from "../../../constants";

export const fetchPatientsList = async (wardId, offset, limit = 3) => {
  return await axios.get(GET_PATIENT_LIST_URL.replace("{wardId}", wardId), {
    params: { offset, limit },
  });
};

export const fetchPatientsListBySearch = async (
  wardId,
  searchKeys,
  searchValue,
  offset,
  limit = 3
) => {
  try {
    return await axios.get(
      GET_SEARCH_PATIENT_LIST_URL.replace("{wardId}", wardId),
      {
        params: {
          offset,
          limit,
          searchKeys,
          searchValue,
        },
        paramsSerializer: {
          indexes: null,
        },
      }
    );
  } catch (e) {
    return e;
  }
};

export const bookmarkPatient = async (patientDetails) => {
  return await axios.post(BOOKMARK_PATIENT_BASE_URL, patientDetails);
};
