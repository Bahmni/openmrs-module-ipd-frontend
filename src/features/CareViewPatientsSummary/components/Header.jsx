import {
  epochTo12HourTimeFormat,
  epochTo24HourTimeFormat,
} from "../../../utils/DateTimeUtils";
import React, { useContext } from "react";
import propTypes from "prop-types";
import { CareViewContext } from "../../../context/CareViewContext";

export const Header = ({ timeframeLimitInHours, navHourEpoch }) => {
  const { ipdConfig } = useContext(CareViewContext);
  const { enable24HourTime = {} } = ipdConfig;
  return (
    <tr className="patient-row-container">
      <td
        className="patient-details-container"
        key="patient-details-header"
        data-testid="patient-details-header"
      ></td>
      {Array.from({ length: timeframeLimitInHours }, (_, i) => {
        const startTime = navHourEpoch.startHourEpoch + i * 3600;
        if (startTime < navHourEpoch.endHourEpoch) {
          let headerKey = `slot-details-header-${i}`;
          return (
            <td
              className="slot-details-header"
              key={headerKey}
              data-testid={headerKey}
            >
              <div className="time" data-testid={`time-frame-${i}`}>
                {enable24HourTime
                  ? epochTo24HourTimeFormat(startTime, true)
                  : epochTo12HourTimeFormat(startTime, true)}
              </div>
            </td>
          );
        }
      })}
    </tr>
  );
};

Header.propTypes = {
  timeframeLimitInHours: propTypes.number.isRequired,
  navHourEpoch: propTypes.object.isRequired,
};
