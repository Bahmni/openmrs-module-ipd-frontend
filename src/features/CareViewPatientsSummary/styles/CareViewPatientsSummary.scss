.patient-row-container {
  display: flex;

  .patient-details-container {
    box-sizing: border-box;
    width: 420px;
    position: sticky;
    left: 0;
    line-height: 1.8;
    display: flex;
    border-right: 1px solid #8d8d8d;
    box-shadow: 6px 2px 4px 0 rgba(0, 0, 0, 0.1);
    padding: 8px;

    a {
      color: #0f62fe;
    }

    .care-view-patient-details {
      flex: 1;
    }
    .admission-details {
      display: flex;
      gap: 8px;
      align-items: center;
    }
    .separator {
      padding: 0 8px;
    }
  }

  .slot-details-header {
    width: 550px;
    padding: 10px 0;
    background-color: #f4f4f4;
    vertical-align: top;
    .time {
      padding-left: 10px;
      font-weight: 500;
    }
  }

  .slot-details-container {
    width: 550px;
    padding: 20px 0;
    vertical-align: top;

    .slot-details {
      display: flex;

      .logo {
        display: flex;
        align-items: start;
        justify-content: space-evenly;
        padding-left: 5px;
        width: 50px;
      }

      .drug-details-wrapper {
        display: flex;
        flex-direction: column;
        padding: 0 0 20px 20px;

        .drug-details {
          display: flex;
          margin-top: 5px;

          .drug-detail {
            margin: 2px;
            white-space: break-spaces;
          }
        }
      }
    }
  }
}

.care-view-table-wrapper {
  text-align: -webkit-center;
}

.care-view-patient-table {
  display: block;
  max-height: 500px;
  max-width: fit-content;
  overflow: scroll;
  margin: 8px 0;
  border: 1px solid #8d8d8d;

  tbody tr:nth-child(odd) td:first-child {
    background-color: #f4f4f4;
  }

  tbody tr:nth-child(even) td:first-child {
    background-color: white;
  }

  tbody {
    tr {
      border-bottom: 0;

      &:nth-child(odd) {
        background-color: #f4f4f4;
      }

      td {
        border: 0;
        padding: 0;
        border-right: 1px solid #8d8d8d;

        &:last-child {
          border-right: 0;
        }
      }
    }

    tr:first-child {
      td:first-child {
        border-right: 1px solid #8d8d8d;
        box-shadow: 6px 2px 4px 0 rgba(0, 0, 0, 0.1);
      }
    }
  }
}
