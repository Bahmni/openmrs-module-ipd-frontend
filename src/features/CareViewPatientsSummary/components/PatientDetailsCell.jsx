import React, { useContext, useState } from "react";
import { bookmarkPatient } from "../../CareViewPatients/utils/CareViewPatientsUtils";
import {
  BookmarkAdd20,
  BookmarkFilled20,
  HospitalBed16,
} from "@carbon/icons-react";
import { Link } from "carbon-components-react";
import { FormattedMessage } from "react-intl";
import propTypes from "prop-types";
import PropTypes from "prop-types";
import { CareViewContext } from "../../../context/CareViewContext";
import {
  currentShiftHoursArray,
  getDateTime,
} from "../../DisplayControls/DrugChart/utils/DrugChartUtils";

export const PatientDetailsCell = (props) => {
  const { patientDetails, bedDetails } = props;
  const { person, uuid } = patientDetails;
  const { ipdConfig, provider } = useContext(CareViewContext);
  const [bookmark, setBookmark] = useState(true);
  const { shiftDetails: shiftConfig = {} } = ipdConfig;

  const getCurrentShiftTimes = (shiftConfig) => {
    const { currentShiftHoursArray: currentShift } = currentShiftHoursArray(
      new Date(),
      shiftConfig
    );
    const firstHour = currentShift[0];
    const lastHour = currentShift[currentShift.length - 1];

    let startDateTime = getDateTimeForHour(firstHour);
    let endDateTime = getDateTimeForHour(lastHour + 1);

    if (lastHour < firstHour) {
      const currentDate = new Date();
      const currentHour = currentDate.getHours();

      if (currentHour > 12) {
        currentDate.setDate(currentDate.getDate() + 1);
        endDateTime = getDateTimeForHour(lastHour + 1, currentDate);
      } else {
        currentDate.setDate(currentDate.getDate() - 1);
        startDateTime = getDateTimeForHour(firstHour, currentDate);
      }
    }

    return { startDateTime, endDateTime };
  };

  const getDateTimeForHour = (hour, date = new Date()) => {
    return getDateTime(date, hour) / 1000;
  };
  const handleBookmarkClick = async (uuid) => {
    const { startDateTime, endDateTime } = getCurrentShiftTimes(shiftConfig);
    const patientData = {
      patientUuid: uuid,
      careTeamParticipantsRequest: [
        {
          providerUuid: provider.uuid,
          startTime: startDateTime,
          endTime: endDateTime,
        },
      ],
    };
    const response = await bookmarkPatient(patientData);
    if (response.status === 200) {
      setBookmark(!bookmark);
    }
  };

  return (
    <td className={"patient-details-container"}>
      <div className={"care-view-patient-details"}>
        <div className={"admission-details"}>
          <HospitalBed16 />|<span>{bedDetails.bedNumber}</span>|
          <Link href={"#"} inline={true}>
            <span>{patientDetails.display.split(" ")[0]}</span>
          </Link>
        </div>
        <div>
          <FormattedMessage id={"PATIENT"} defaultMessage={"Patient"} />:{" "}
          <span>{person.display}</span>&nbsp;(
          <span>{person.gender}</span>)<span className={"separator"}>|</span>
          <span>{person.age}</span>
          <FormattedMessage id={"AGE_YEARS_LABEL"} defaultMessage={"yrs"} />
        </div>
      </div>
      <div onClick={() => handleBookmarkClick(uuid)}>
        {bookmark ? <BookmarkAdd20 /> : <BookmarkFilled20 />}
      </div>
    </td>
  );
};

PatientDetailsCell.propTypes = {
  patientDetails: {
    person: PropTypes.object.isRequired,
    uuid: PropTypes.string.isRequired,
  },
  bedDetails: propTypes.object.isRequired,
};
