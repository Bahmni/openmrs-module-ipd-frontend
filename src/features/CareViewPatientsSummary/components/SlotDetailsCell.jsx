import {
  getColumnData,
  getTaskColumnData,
} from "../../CareViewSummary/utils/CareViewSummary";
import React, { useContext } from "react";
import Clock from "../../../icons/clock.svg";
import { epochTo24HourTimeFormat } from "../../../utils/DateTimeUtils";
import { getAdministrationStatus } from "../../../utils/CommonUtils";
import SVGIcon from "../../SVGIcon/SVGIcon";
import { CareViewContext } from "../../../context/CareViewContext";
import propTypes from "prop-types";

export const SlotDetailsCell = ({
  uuid,
  slotDetails,
  timeframeLimitInHours,
  navHourEpoch,
  nonMedicationDetails,
  filterValue,
}) => {
  const columns = [];
  const { ipdConfig } = useContext(CareViewContext);

  const patientSlotDetail = slotDetails?.find(
    (slotDetail) => slotDetail.patientUuid === uuid
  );

  for (let i = 0; i < timeframeLimitInHours; i++) {
    const startTime = navHourEpoch.startHourEpoch + i * 3600;
    if (startTime < navHourEpoch.endHourEpoch) {
      const endTime = startTime + 3600;
      let columnData, slotColumnData, taskColumnData;
      slotColumnData =
        patientSlotDetail &&
        (filterValue.id === "allTasks" || filterValue.id === "medicationTasks")
          ? patientSlotDetail?.prescribedOrderSlots.flatMap((slot) =>
              getColumnData(slot, startTime, endTime)
            )
          : [];
      taskColumnData =
        nonMedicationDetails &&
        nonMedicationDetails[uuid] &&
        (filterValue.id === "allTasks" ||
          filterValue.id === "nonMedicationTasks")
          ? getTaskColumnData(nonMedicationDetails[uuid], startTime, endTime)
          : [];
      columnData = slotColumnData.concat(...taskColumnData);
      if (columnData) {
        columns.push({ startTime, endTime, columnData });
      } else {
        columns.push({ startTime, endTime, columnData: [] });
      }
    }
  }

  const renderStatusIcon = (slot) => {
    const { drugChart = {} } = ipdConfig;
    const { startTime, status, medicationAdministration } = slot;
    let administrationStatus = getAdministrationStatus(
      medicationAdministration,
      status,
      startTime,
      drugChart,
      slot
    );
    return (
      <div className="status-icon">
        <SVGIcon iconType={administrationStatus} />
      </div>
    );
  };

  const renderColumnData = (columnData) => {
    columnData.sort((a, b) => a.startTime - b.startTime);

    return columnData.map((slotItem) => {
      if (slotItem.isNonMedication) {
        return (
          <div className="slot-details" key={`${slotItem.uuid}`}>
            <div className="logo">
              <div className="status-icon">
                <SVGIcon iconType={"Pending"} />
              </div>
              <Clock className="clock-icon" />
            </div>
            <span>
              {epochTo24HourTimeFormat(slotItem.requestedStartTime / 1000)}
            </span>
            <div className="drug-details-wrapper">
              <span>{slotItem.name}</span>
              <div className="drug-details" data-testid="drug-details">
                {slotItem.creator && (
                  <span className="drug-detail">
                    {slotItem.creator.display}
                  </span>
                )}
              </div>
            </div>
          </div>
        );
      } else {
        const { dose, doseUnits, route } = slotItem.order;
        return (
          <div className="slot-details" key={`${slotItem.uuid}`}>
            <div className="logo">
              {renderStatusIcon(slotItem)}
              <Clock className="clock-icon" />
            </div>
            <span>{epochTo24HourTimeFormat(slotItem.startTime)}</span>
            <div className="drug-details-wrapper">
              <span>{slotItem.order.drug.display}</span>
              <div className="drug-details" data-testid="drug-details">
                {dose && <span className="drug-detail">{dose}</span>}
                {doseUnits && (
                  <span className="drug-detail">{doseUnits?.display}</span>
                )}
                {route && (
                  <span className="drug-detail">
                    {route?.display && `  |  ${route.display}`}
                  </span>
                )}
              </div>
            </div>
          </div>
        );
      }
    });
  };

  return columns.map(({ columnData }, index) => (
    <td className="slot-details-container" key={index}>
      {renderColumnData(columnData)}
    </td>
  ));
};

SlotDetailsCell.propTypes = {
  uuid: propTypes.string.isRequired,
  slotDetails: propTypes.object.isRequired,
  timeframeLimitInHours: propTypes.number.isRequired,
  navHourEpoch: propTypes.object.isRequired,
};
