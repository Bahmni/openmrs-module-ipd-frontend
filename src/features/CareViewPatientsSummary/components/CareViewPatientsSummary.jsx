import React, { useContext, useEffect, useState } from "react";
import propTypes from "prop-types";
import "../styles/CareViewPatientsSummary.scss";
import { getSlotsForPatients } from "../../CareViewSummary/utils/CareViewSummary";
import {
  getPreviousNearbyHourEpoch,
  getTimeInFuture,
} from "../../../utils/DateTimeUtils";
import { CareViewContext } from "../../../context/CareViewContext";
import { PatientDetailsCell } from "./PatientDetailsCell";
import { SlotDetailsCell } from "./SlotDetailsCell";
import { Header } from "./Header";

export const CareViewPatientsSummary = (props) => {
  const [slotDetails, setSlotDetails] = useState([]);
  const { careViewConfig } = useContext(CareViewContext);
  const { patientsSummary } = props;
  const timeframeLimitInHours = careViewConfig.timeframeLimitInHours;
  const currentEpoch = Math.floor(new Date().getTime() / 1000);
  const nearestHourEpoch = getPreviousNearbyHourEpoch(currentEpoch);

  const fetchSlots = async (patients) => {
    const patientUuids = patients.map((patient) => patient.patientDetails.uuid);
    const currentTime = getPreviousNearbyHourEpoch(
      Math.floor(Date.now() / 1000)
    );
    const endTime = getTimeInFuture(currentTime, timeframeLimitInHours);
    const response = await getSlotsForPatients(
      patientUuids,
      currentTime,
      endTime
    );
    setSlotDetails(response);
  };

  useEffect(() => {
    if (patientsSummary.length > 0) {
      fetchSlots(patientsSummary);
    }
  }, [patientsSummary]);

  return (
    <div className={"care-view-table-wrapper"}>
      <table className={"care-view-patient-table"}>
        <tbody>
          <Header
            timeframeLimitInHours={timeframeLimitInHours}
            nearestHourEpoch={nearestHourEpoch}
          />
          {patientsSummary.map((patientSummary, idx) => {
            const { patientDetails, bedDetails } = patientSummary;
            const { uuid } = patientDetails;
            return (
              <tr key={idx} className={"patient-row-container"}>
                <PatientDetailsCell
                  bedDetails={bedDetails}
                  patientDetails={patientDetails}
                />
                <SlotDetailsCell
                  slotDetails={slotDetails}
                  uuid={uuid}
                  timeframeLimitInHours={timeframeLimitInHours}
                  nearestHourEpoch={nearestHourEpoch}
                />
              </tr>
            );
          })}
        </tbody>
      </table>
    </div>
  );
};

CareViewPatientsSummary.propTypes = {
  patientsSummary: propTypes.array,
};
