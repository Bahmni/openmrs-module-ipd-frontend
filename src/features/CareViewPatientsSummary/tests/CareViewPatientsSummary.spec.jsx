import React from "react";
import { render, screen, waitFor } from "@testing-library/react";
import { CareViewPatientsSummary } from "../components/CareViewPatientsSummary";
import { mockPatientsList } from "./CareViewPatientsSummaryMock";
import { CareViewContext } from "../../../context/CareViewContext";
import MockDate from "mockdate";
import { mockConfig } from "../../../utils/CommonUtils";
import {
  mockColumnData,
  mockSlotsData,
} from "../../CareViewSummary/tests/CareViewSummaryMock";
import "@testing-library/jest-dom/extend-expect";

const mockContext = {
  careViewConfig: { timeframeLimitInHours: 2 },
  ipdConfig: mockConfig,
};
const mockNavHourEpoch = {
  startHourEpoch: 1704110400,
  endHourEpoch: 1704117600,
};

const mockGetSlotsForPatients = jest.fn();
const mockGetColumnData = jest.fn();
jest.mock("../../CareViewSummary/utils/CareViewSummary", () => {
  return {
    getSlotsForPatients: () => mockGetSlotsForPatients(),
    getColumnData: () => mockGetColumnData(),
  };
});

describe("CareViewPatientsSummary", function () {
  afterEach(() => {
    MockDate.reset();
  });

  beforeEach(() => {
    jest.clearAllMocks();
    MockDate.set("2023-01-01T12:00:00");
    mockGetSlotsForPatients.mockReturnValue(mockSlotsData);
    mockGetColumnData.mockReturnValue(mockColumnData);
  });

  it("should match snapshot", () => {
    const { container } = render(
      <CareViewContext.Provider value={mockContext}>
        <CareViewPatientsSummary
          patientsSummary={mockPatientsList.admittedPatients}
          navHourEpoch={mockNavHourEpoch}
        />
      </CareViewContext.Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it("should fetch slot details on initial render", () => {
    render(
      <CareViewContext.Provider value={mockContext}>
        <CareViewPatientsSummary
          patientsSummary={mockPatientsList.admittedPatients}
          navHourEpoch={mockNavHourEpoch}
        />
      </CareViewContext.Provider>
    );

    expect(mockGetSlotsForPatients).toHaveBeenCalled();
  });

  it("renders table headers correctly", async () => {
    const { queryByTestId } = render(
      <CareViewContext.Provider value={mockContext}>
        <CareViewPatientsSummary
          patientsSummary={mockPatientsList.admittedPatients}
          navHourEpoch={mockNavHourEpoch}
        />
      </CareViewContext.Provider>
    );

    await waitFor(() => {
      expect(queryByTestId("slot-details-header-0")).toBeTruthy();
      expect(queryByTestId("time-frame-0")).toHaveTextContent("12:00");
      expect(queryByTestId("time-frame-1")).toHaveTextContent("13:00");
      expect(queryByTestId("time-frame-2")).toBeNull();
    });
  });

  it("renders patient details correctly", async () => {
    const { queryByText } = render(
      <CareViewContext.Provider value={mockContext}>
        <CareViewPatientsSummary
          patientsSummary={mockPatientsList.admittedPatients}
          navHourEpoch={mockNavHourEpoch}
        />
      </CareViewContext.Provider>
    );

    await waitFor(() => {
      expect(queryByText("PT51140")).toBeTruthy();
      expect(
        queryByText("AnnonFN-Jcilhyxuen AnnonMN-Dylkrgbpwo AnnonLN-Gkksnhzbeu")
      ).toBeTruthy();
      expect(queryByText("A-6")).toBeTruthy();
      expect(queryByText("14")).toBeTruthy();
    });
  });

  it("renders slot details correctly", async () => {
    const { queryAllByText, queryAllByTestId } = render(
      <CareViewContext.Provider value={mockContext}>
        <CareViewPatientsSummary
          patientsSummary={mockPatientsList.admittedPatients}
          navHourEpoch={mockNavHourEpoch}
        />
      </CareViewContext.Provider>
    );

    await waitFor(() => {
      expect(
        queryAllByText("Amoxicillin/Clavulanic Acid 1000 mg Tablet")
      ).toBeTruthy();
      expect(queryAllByTestId("drug-details")[0]).toHaveTextContent(
        "2Tablet(s) | Oral"
      );
    });
  });

  it("should new treatments notification be present under patient details", async () => {
    render(
      <CareViewContext.Provider value={mockContext}>
        <CareViewPatientsSummary
          patientsSummary={mockPatientsList.admittedPatients}
          navHourEpoch={mockNavHourEpoch}
        />
      </CareViewContext.Provider>
    );
    const careViewPatientDetails = screen.getAllByText(/New treatment/i);
    expect(careViewPatientDetails.length).toBe(1);
  });
});
