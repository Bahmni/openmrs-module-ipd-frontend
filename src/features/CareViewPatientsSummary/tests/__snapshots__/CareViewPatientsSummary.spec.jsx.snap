// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CareViewPatientsSummary should match snapshot 1`] = `
<div>
  <div
    class="care-view-table-wrapper"
  >
    <table
      class="care-view-patient-table"
    >
      <tbody>
        <tr
          class="patient-row-container"
        >
          <td
            class="patient-details-container"
            data-testid="patient-details-header"
          />
          <td
            class="slot-details-header"
            data-testid="slot-details-header-0"
          >
            <div
              class="time"
              data-testid="time-frame-0"
            >
              12:00
            </div>
          </td>
          <td
            class="slot-details-header"
            data-testid="slot-details-header-1"
          >
            <div
              class="time"
              data-testid="time-frame-1"
            >
              13:00
            </div>
          </td>
        </tr>
        <tr
          class="patient-row-container"
        >
          <td
            class="patient-details-container"
          >
            <div
              class="care-view-patient-details"
            >
              <div
                class="admission-details"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  focusable="false"
                  height="16"
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 32 32"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M25,16H17a2.0023,2.0023,0,0,0-2,2v6H4V14H2V30H4V26H28v4h2V21A5.0059,5.0059,0,0,0,25,16Zm3,8H17V18h8a3.0033,3.0033,0,0,1,3,3Z"
                  />
                  <path
                    d="M9.5 17A1.5 1.5 0 118 18.5 1.5017 1.5017 0 019.5 17m0-2A3.5 3.5 0 1013 18.5 3.5 3.5 0 009.5 15zM21 6L17 6 17 2 15 2 15 6 11 6 11 8 15 8 15 12 17 12 17 8 21 8 21 6z"
                  />
                </svg>
                |
                <span>
                  A-6
                </span>
                |
                <a
                  class="bx--link bx--link--inline"
                  href="#"
                >
                  <span>
                    PT51140
                  </span>
                </a>
              </div>
              <div>
                Patient
                :
                 
                <span>
                  AnnonFN-Jcilhyxuen AnnonMN-Dylkrgbpwo AnnonLN-Gkksnhzbeu
                </span>
                 (
                <span>
                  M
                </span>
                )
                <span
                  class="separator"
                >
                  |
                </span>
                <span>
                  14
                </span>
                yrs
              </div>
            </div>
            <div>
              <svg
                aria-hidden="true"
                fill="currentColor"
                focusable="false"
                height="20"
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 32 32"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M24,16V26.7515l-7.0962-3.5894L16,22.7051l-.9009.456L8,26.748V4H18V2H8A2,2,0,0,0,6,4V30l10-5.0537L26,30V16Z"
                />
                <path
                  d="M26 6L26 2 24 2 24 6 20 6 20 8 24 8 24 12 26 12 26 8 30 8 30 6 26 6z"
                />
              </svg>
            </div>
          </td>
          <td
            class="slot-details-container"
          />
          <td
            class="slot-details-container"
          />
        </tr>
        <tr
          class="patient-row-container"
        >
          <td
            class="patient-details-container"
          >
            <div
              class="care-view-patient-details"
            >
              <div
                class="admission-details"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  focusable="false"
                  height="16"
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 32 32"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M25,16H17a2.0023,2.0023,0,0,0-2,2v6H4V14H2V30H4V26H28v4h2V21A5.0059,5.0059,0,0,0,25,16Zm3,8H17V18h8a3.0033,3.0033,0,0,1,3,3Z"
                  />
                  <path
                    d="M9.5 17A1.5 1.5 0 118 18.5 1.5017 1.5017 0 019.5 17m0-2A3.5 3.5 0 1013 18.5 3.5 3.5 0 009.5 15zM21 6L17 6 17 2 15 2 15 6 11 6 11 8 15 8 15 12 17 12 17 8 21 8 21 6z"
                  />
                </svg>
                |
                <span>
                  C-1
                </span>
                |
                <a
                  class="bx--link bx--link--inline"
                  href="#"
                >
                  <span>
                    PT49722
                  </span>
                </a>
              </div>
              <div>
                Patient
                :
                 
                <span>
                  AnnonFN-Eifbqbgbnm AnnonMN-Vrystpswee AnnonLN-Idseppecxk
                </span>
                 (
                <span>
                  M
                </span>
                )
                <span
                  class="separator"
                >
                  |
                </span>
                <span>
                  10
                </span>
                yrs
              </div>
            </div>
            <div>
              <svg
                aria-hidden="true"
                fill="currentColor"
                focusable="false"
                height="20"
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 32 32"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M24,16V26.7515l-7.0962-3.5894L16,22.7051l-.9009.456L8,26.748V4H18V2H8A2,2,0,0,0,6,4V30l10-5.0537L26,30V16Z"
                />
                <path
                  d="M26 6L26 2 24 2 24 6 20 6 20 8 24 8 24 12 26 12 26 8 30 8 30 6 26 6z"
                />
              </svg>
            </div>
          </td>
          <td
            class="slot-details-container"
          />
          <td
            class="slot-details-container"
          />
        </tr>
        <tr
          class="patient-row-container"
        >
          <td
            class="patient-details-container"
          >
            <div
              class="care-view-patient-details"
            >
              <div
                class="admission-details"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  focusable="false"
                  height="16"
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 32 32"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M25,16H17a2.0023,2.0023,0,0,0-2,2v6H4V14H2V30H4V26H28v4h2V21A5.0059,5.0059,0,0,0,25,16Zm3,8H17V18h8a3.0033,3.0033,0,0,1,3,3Z"
                  />
                  <path
                    d="M9.5 17A1.5 1.5 0 118 18.5 1.5017 1.5017 0 019.5 17m0-2A3.5 3.5 0 1013 18.5 3.5 3.5 0 009.5 15zM21 6L17 6 17 2 15 2 15 6 11 6 11 8 15 8 15 12 17 12 17 8 21 8 21 6z"
                  />
                </svg>
                |
                <span>
                  ICU1
                </span>
                |
                <a
                  class="bx--link bx--link--inline"
                  href="#"
                >
                  <span>
                    ET55749
                  </span>
                </a>
              </div>
              <div>
                Patient
                :
                 
                <span>
                  user three
                </span>
                 (
                <span>
                  F
                </span>
                )
                <span
                  class="separator"
                >
                  |
                </span>
                <span>
                  30
                </span>
                yrs
              </div>
            </div>
            <div>
              <svg
                aria-hidden="true"
                fill="currentColor"
                focusable="false"
                height="20"
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 32 32"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M24,16V26.7515l-7.0962-3.5894L16,22.7051l-.9009.456L8,26.748V4H18V2H8A2,2,0,0,0,6,4V30l10-5.0537L26,30V16Z"
                />
                <path
                  d="M26 6L26 2 24 2 24 6 20 6 20 8 24 8 24 12 26 12 26 8 30 8 30 6 26 6z"
                />
              </svg>
            </div>
          </td>
          <td
            class="slot-details-container"
          />
          <td
            class="slot-details-container"
          />
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
